File: LmsApplication.CourseBoardModule.Data/Models/CommentModel.cs
using LmsApplication.Core.Shared.Models;
using LmsApplication.CourseBoardModule.Data.Entities;

namespace LmsApplication.CourseBoardModule.Data.Models;

public class CommentModel
{
    public required Guid Id { get; set; }
    
    public required string Content { get; set; }
    
    public required Guid PostId { get; set; }
    
    public required UserExchangeModel Author { get; set; }
    
    public required ReactionModel Reactions { get; set; }
    
    public required ReactionType? CurrentUserReaction { get; set; }
    
    public required DateTime CreatedAt { get; set; }
    
    public required DateTime? UpdatedAt { get; set; }
}

File: LmsApplication/LmsApplication.Frontend/src/app/types/course-board/comment-model.ts
import { UserModel } from "../users/user-model";
import { ReactionModel } from "./reaction-model";
import { ReactionType } from "./reaction-type";

export interface CommentModel {
  id: string;
  content: string;
  postId: string;
  author: UserModel;
  reactions: ReactionModel;
  currentUserReaction: ReactionType | null;
  createdAt: Date;
  updatedAt: Date | null;
}
